plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/meta/CubridMetaModel.java

```diff
@@ -56,106 +68,348 @@ public String getTableOrViewName(GenericTableBase base)
+            throws DBException {
         List<GenericSchema> users = new ArrayList<>();
-        try {
-            final JDBCPreparedStatement dbStat = session.prepareStatement("select * from db_user");
+        try (JDBCPreparedStatement dbStat = session.prepareStatement("select * from db_user")) {
             dbStat.executeStatement();
-            JDBCResultSet dbResult = dbStat.getResultSet();
-
-            while (dbResult.next()) {
-                String name = JDBCUtils.safeGetStringTrimmed(dbResult, "name");
-                String description = JDBCUtils.safeGetStringTrimmed(dbResult, "comment");
-                CubridUser user = new CubridUser(dataSource, name, description);
-                users.add(user);
+            try (JDBCResultSet dbResult = dbStat.getResultSet()) {
+                while (dbResult.next()) {
+                    String name = JDBCUtils.safeGetStringTrimmed(dbResult, CubridConstants.NAME);
+                    String description = JDBCUtils.safeGetStringTrimmed(dbResult, CubridConstants.COMMENT);
+                    CubridUser user = new CubridUser(dataSource, name, description);
+                    users.add(user);
+	            }
             }
         } catch (SQLException e) {
-        	log.error("Cannot load user", e);
+            log.error("Cannot load user", e);
         }
         return users;
     }
 
+    @NotNull
     @Override
     public JDBCStatement prepareTableLoadStatement(
             @NotNull JDBCSession session,
             @NotNull GenericStructContainer owner,
             @Nullable GenericTableBase object,
             @Nullable String objectName)
-            throws SQLException
-    {
+            throws SQLException {
         String sql = "select a.*,a.class_name as TABLE_NAME, case when class_type = 'CLASS' then 'TABLE' \r\n"
                 + "when class_type = 'VCLASS' then 'VIEW' end as TABLE_TYPE, \r\n"
-                + "b.current_val from db_class a LEFT JOIN db_serial b on \r\n"
-                + "a.class_name = b.class_name "
+                + "a.comment as REMARKS, b.current_val from db_class a LEFT JOIN \r\n"
+                + "db_serial b on a.class_name = b.class_name "
                 + "where a.owner_name = ?";
-
         final JDBCPreparedStatement dbStat = session.prepareStatement(sql);
         dbStat.setString(1, owner.getName());
         return dbStat;
     }
 
+    @NotNull
     @Override
     public JDBCStatement prepareTableColumnLoadStatement(
             @NotNull JDBCSession session,
             @NotNull GenericStructContainer owner,
             @Nullable GenericTableBase forTable)
-            throws SQLException
-    {
-        return session.getMetaData().getColumns(null, null, this.getTableOrViewName(forTable), null).getSourceStatement();
+            throws SQLException {
+        Boolean multiSchema = forTable != null ? ((CubridDataSource) forTable.getDataSource()).getSupportMultiSchema() : false;

```

Comment:
forTable != null && ((CubridDataSource) forTable.getDataSource()).getSupportMultiSchema();

Reply:
I have fixed this.

Reply:
I have fixed this.